In [3]:
import pandas as pd
import numpy as np

pd.set_option('display.max_columns', None)

1. Apertura del csv.

In [4]:
df_datos = pd.read_csv("human_cognitive_performance.csv")

df_datos.head(3)

User_ID  Age  Gender  Sleep_Duration  Stress_Level       Diet_Type  \
0      U1   57  Female             6.5             3  Non-Vegetarian   
1      U2   39  Female             7.6             9  Non-Vegetarian   
2      U3   26    Male             8.2             6      Vegetarian   

   Daily_Screen_Time Exercise_Frequency  Caffeine_Intake  Reaction_Time  \
0                6.5             Medium               41         583.33   
1               10.8               High              214         368.24   
2                5.7                Low              429         445.21   

   Memory_Test_Score  Cognitive_Score  AI_Predicted_Score  
0                 65            36.71               39.77  
1                 58            54.35               57.68  
2                 49            32.57               29.54

2.

In [4]:
df_datos.tail(3)

User_ID  Age  Gender  Sleep_Duration  Stress_Level       Diet_Type  \
79997  U79998   32  Female             7.3            10  Non-Vegetarian   
79998  U79999   34    Male             8.7             8      Vegetarian   
79999  U80000   41  Female             7.9             7      Vegetarian   

       Daily_Screen_Time Exercise_Frequency  Caffeine_Intake  Reaction_Time  \
79997                7.5                Low              462         249.57   
79998                7.2             Medium               96         319.48   
79999                2.0             Medium               27         263.21   

       Memory_Test_Score  Cognitive_Score  AI_Predicted_Score  
79997                 72            63.51               67.35  
79998                 82            81.43               77.95  
79999                 42            80.39               84.45

In [5]:
df_datos.shape

(80000, 13)

In [6]:
df_datos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 80000 entries, 0 to 79999
Data columns (total 13 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   User_ID             80000 non-null  object 
 1   Age                 80000 non-null  int64  
 2   Gender              80000 non-null  object 
 3   Sleep_Duration      80000 non-null  float64
 4   Stress_Level        80000 non-null  int64  
 5   Diet_Type           80000 non-null  object 
 6   Daily_Screen_Time   80000 non-null  float64
 7   Exercise_Frequency  80000 non-null  object 
 8   Caffeine_Intake     80000 non-null  int64  
 9   Reaction_Time       80000 non-null  float64
 10  Memory_Test_Score   80000 non-null  int64  
 11  Cognitive_Score     80000 non-null  float64
 12  AI_Predicted_Score  80000 non-null  float64
dtypes: float64(5), int64(4), object(4)
memory usage: 7.9+ MB


¿Hay duplicados en los ids?

In [13]:
df_datos['User_ID'].duplicated().sum()

0

In [14]:
df_datos["User_ID"].nunique()

80000

NO HAY VALORES DUPLICADOS

In [16]:
df_datos.describe().T

count        mean         std    min     25%     50%  \
Age                 80000.0   38.525525   12.101876   18.0   28.00   39.00   
Sleep_Duration      80000.0    7.005332    1.734435    4.0    5.50    7.00   
Stress_Level        80000.0    5.493762    2.865308    1.0    3.00    5.00   
Daily_Screen_Time   80000.0    6.504646    3.167072    1.0    3.80    6.50   
Caffeine_Intake     80000.0  248.988213  144.541990    0.0  123.00  249.00   
Reaction_Time       80000.0  399.973579  115.369329  200.0  300.15  400.36   
Memory_Test_Score   80000.0   69.498350   17.305659   40.0   55.00   70.00   
Cognitive_Score     80000.0   58.172395   23.058522    0.0   40.91   58.36   
AI_Predicted_Score  80000.0   58.121357   23.119598    0.0   40.85   58.36   

                       75%     max  
Age                  49.00   59.00  
Sleep_Duration        8.50   10.00  
Stress_Level          8.00   10.00  
Daily_Screen_Time     9.20   12.00  
Caffeine_Intake     375.00  499.00  
Reaction_Time       499.25  599.99  
Memory_Test_Score    85.00   99.00  
Cognitive_Score      75.83  100.00  
AI_Predicted_Score   75.89  100.00

*AGE*

In [17]:
df_datos["Age"].value_counts()

Age
40    1979
50    1972
32    1963
29    1961
49    1959
44    1956
37    1955
43    1944
18    1941
33    1938
28    1928
19    1925
54    1924
58    1921
34    1921
55    1920
47    1920
48    1918
35    1915
56    1915
27    1914
22    1909
39    1909
51    1902
25    1901
30    1893
46    1888
53    1887
57    1887
31    1878
38    1877
24    1875
42    1872
59    1868
41    1859
21    1858
36    1853
20    1848
52    1848
26    1847
23    1829
45    1823
Name: count, dtype: int64

### Averiguamos edad mínima y edad máxima y promedio

In [10]:
df_datos["Age"].agg(['min', 'max', 'mean'])

min     18.000000
max     59.000000
mean    38.525525
Name: Age, dtype: float64

**REALIZAR HISTOGRAMA PARA CLASIFICAR POR GRUPOS**

In [8]:
df_datos.describe(include="O").T

count unique             top   freq
User_ID             80000  80000              U1      1
Gender              80000      3          Female  38404
Diet_Type           80000      3  Non-Vegetarian  47986
Exercise_Frequency  80000      3          Medium  31990

*GENDER*

In [20]:
df_datos["Gender"].unique()

array(['Female', 'Male', 'Other'], dtype=object)

In [23]:
df_datos["Gender"].value_counts()

Gender
Female    38404
Male      38322
Other      3274
Name: count, dtype: int64

**Investigar Other**

*SLEEP DURATION*

In [24]:
df_datos["Sleep_Duration"].unique()

array([ 6.5,  7.6,  8.2,  7.8,  9.7,  7.4,  6.7,  6. ,  5.1,  7.9,  7.3,
        9.5,  9.3,  4.6,  6.3,  7.5,  5.7,  5.8,  8.7,  8.4,  4.1,  6.9,
        5.2,  9.4,  6.8,  4.8,  8.6,  6.6,  9.1,  8.1,  7.1,  8.3,  6.4,
        4.2,  9. ,  9.9,  4.7,  8.5,  8. ,  8.9,  4.5,  9.6,  5.6, 10. ,
        9.2,  5.9,  5.3,  8.8,  5. ,  5.5,  4.3,  7.2,  4.4,  4.9,  7.7,
        6.1,  9.8,  4. ,  5.4,  6.2,  7. ])

In [25]:
df_datos["Sleep_Duration"].nunique()

61

In [26]:
df_datos["Sleep_Duration"].value_counts()

Sleep_Duration
9.9     1410
4.4     1408
6.8     1407
7.9     1407
5.4     1394
        ... 
4.8     1263
4.9     1259
5.6     1246
4.0      678
10.0     669
Name: count, Length: 61, dtype: int64

Sólo 669 duermen 10 horas 1410 duermen 9.9 horas

### Averiguamos horas de sueño mínimas, máximas y promedio

In [11]:
df_datos["Sleep_Duration"].agg(['min', 'max', 'mean'])

min      4.000000
max     10.000000
mean     7.005332
Name: Sleep_Duration, dtype: float64

**¿CLASIFICAR POR RANGOS?, VER VALORES ATÍPICOS DE HORAS DE SUEÑO**

*STRESS LEVEL*

In [20]:
df_datos['Stress_Level'].unique()

array([ 3,  9,  6,  2, 10,  1,  8,  5,  7,  4], dtype=int64)

In [30]:
df_datos['Stress_Level'].nunique()

10

In [29]:
df_datos['Stress_Level'].value_counts()

Stress_Level
3     8103
8     8092
5     8070
7     8027
6     8020
4     8019
10    7969
1     7969
2     7931
9     7800
Name: count, dtype: int64

### Averiguamos los valores máximos, mínimos y medios registrados de 'nivel de estrés'

In [12]:
df_datos["Stress_Level"].agg(['min', 'max', 'mean'])

min      1.000000
max     10.000000
mean     5.493762
Name: Stress_Level, dtype: float64

**RELACIÓN ESTRÉS CON TIEMPO DE REACCIÓN, HORAS DE SUEÑO, DIETA, EDAD, HORAS EN PANTALLA...**

In [ ]:
df_datos['Diet_Type'].unique()

array(['Non-Vegetarian', 'Vegetarian', 'Vegan'], dtype=object)

In [33]:
df_datos['Diet_Type'].value_counts()

Diet_Type
Non-Vegetarian    47986
Vegetarian        24136
Vegan              7878
Name: count, dtype: int64

**RELACIÓN CON NIVEL DE ESTRÉS** , **R CON TESTS COG** , **R CON GENERO, EDAD, DURACIÓN DEL SUEÑO**
¿Influye la dieta en la capacidad cognitiva? ¿Varían los hábitos cotidianos o el estrés según la dieta que se tenga?

*DAILY SCREEN TIME*

In [22]:
df_datos['Daily_Screen_Time'].unique()

array([ 6.5, 10.8,  5.7,  8.3, 11.3,  6.6,  1.4,  6.8, 10.2,  9.1, 11.7,
        2.6,  5.5,  3.5,  6.1,  8.7,  3.1,  4.8,  5.6, 10.7,  4. ,  9.4,
       11. ,  7.5, 11.6,  4.9,  8.5,  2.2,  5. , 10.4,  1.3, 10.9, 10.3,
        5.2, 11.9,  3.7,  3.2,  4.1,  9.8,  5.1,  1.6,  5.8,  4.4,  4.6,
        7.2,  4.7,  7.1,  9.5,  1.7,  8.6, 10.6,  3.6,  4.2,  5.9,  4.5,
        3.4,  8. ,  1.8,  3. , 11.1,  9.2,  8.4,  9.7,  2.7,  2.4,  2.3,
        8.2,  5.3,  1.2,  2.8,  2.1,  7. , 10.1,  8.1,  9.9,  6.2,  8.8,
       11.5,  2.5,  7.4,  3.9,  4.3,  7.8, 10. ,  6.4,  1.9,  2.9,  6.3,
        1.1,  6.7,  3.3,  8.9,  3.8,  9.3,  6.9, 10.5,  5.4,  7.7,  1. ,
       11.4, 11.2, 12. ,  7.3,  2. , 11.8,  6. ,  9. ,  9.6,  1.5,  7.9,
        7.6])

In [34]:
df_datos['Daily_Screen_Time'].nunique()

111

In [ ]:
df_datos['Daily_Screen_Time'].value_counts()

Daily_Screen_Time
7.7     796
5.2     795
10.6    789
1.4     776
4.2     776
       ... 
5.4     670
2.5     669
7.5     653
1.0     367
12.0    352
Name: count, Length: 111, dtype: int64

### Averiguamos los horas mínimas, máximas y el promedio registrados

In [13]:
df_datos['Daily_Screen_Time'].agg(['min', 'max', 'mean'])

min      1.000000
max     12.000000
mean     6.504646
Name: Daily_Screen_Time, dtype: float64

**¿HAY VALORES EXTREMOS? ¿PROMEDIO? RELACION CON SUEÑO Y RENDIMIENTO COGNITIVO**

*EXERCISE FREQUENCY*

In [23]:
df_datos['Exercise_Frequency'].unique()

array(['Medium', 'High', 'Low'], dtype=object)

In [39]:
df_datos['Exercise_Frequency'].value_counts()

Exercise_Frequency
Medium    31990
Low       31896
High      16114
Name: count, dtype: int64

**RELACION CON ESTRÉS Y RENDIMIENTO COGNITIVO**

*CAFFEINE INTAKE*

In [40]:
df_datos['Caffeine_Intake'].nunique()

500

In [41]:
df_datos['Caffeine_Intake'].value_counts()

Caffeine_Intake
217    198
76     198
30     196
238    192
46     191
      ... 
264    134
112    134
173    133
373    132
246    132
Name: count, Length: 500, dtype: int64

In [ ]:
df_datos['Caffeine_Intake'].agg(['min', 'max', 'mean']) # miligramos, diferentes bebidas con cafeina

min       0.000000
max     499.000000
mean    248.988213
Name: Caffeine_Intake, dtype: float64

**ESTANDARIZAR UNIDAD DE MEDIDA EN TAZAS**, **¿INGESTA PROMEDIO?** **RELACIONAR CON HORAS DE SUEÑO Y TIEMPO DE REACCIÓN**

*REACTION TIME*

In [42]:
df_datos['Reaction_Time'].nunique() #prueba cognitiva, tiempo de reacción a estimulos, se mide en milisegundos

34595

In [34]:
df_datos['Reaction_Time'].value_counts()

Reaction_Time
260.11    10
597.79     9
567.89     9
424.62     9
316.02     9
          ..
376.12     1
415.01     1
512.33     1
246.85     1
290.17     1
Name: count, Length: 34595, dtype: int64

In [35]:
df_datos['Reaction_Time'].agg(['min', 'max', 'mean'])

min     200.000000
max     599.990000
mean    399.973579
Name: Reaction_Time, dtype: float64

**¿VALORES ATIPICOS (DEMASIADO LENTO O RAPIDO)?, RELACIÓN CON EDAD, PUNTUACIÓN TESTS, ...**

*MEMORY TEST SCORE*

In [44]:
df_datos['Memory_Test_Score'].nunique()

60

In [38]:
df_datos['Memory_Test_Score'].agg(['min', 'max', 'mean'])

min     40.00000
max     99.00000
mean    69.49835
Name: Memory_Test_Score, dtype: float64

**RELACIÓN CON EDAD, DIETA, TIEMPO DE PANTALLA ...**

*COGNITIVE SCORE*

In [46]:
df_datos['Cognitive_Score'].nunique()

9492

In [40]:
df_datos['Cognitive_Score'].agg(['min', 'max', 'mean'])

min       0.000000
max     100.000000
mean     58.172395
Name: Cognitive_Score, dtype: float64

**COMPARAR CON PUNTUACION PROMEDIO, RELACIÓN CON EDAD, DIETA, TIEMPO DE PANTALLA ...**

*AI PREDICTED SCORE*

In [41]:
df_datos['AI_Predicted_Score'].unique()

array([39.77, 57.68, 29.54, ...,  3.98,  8.2 , 15.7 ])

In [42]:
df_datos['AI_Predicted_Score'].agg(['min', 'max', 'mean'])

min       0.000000
max     100.000000
mean     58.121357
Name: AI_Predicted_Score, dtype: float64

**COMPARAR**